src/SFA.DAS.FindEmploymentSchemes.Web/Views/CaseStudies/CaseStudyPage.cshtml

```diff
@@ -1,18 +1,44 @@
-﻿@model CaseStudyPage
+﻿@using SFA.DAS.FindEmploymentSchemes.Contentful.Model.Content
+@model CaseStudyPagePreview
 
 @{
-    ViewData["Title"] = Model.Title.Substring(0, Math.Min(Model.Title.Length, 64));;
+    ViewData["Title"] = Model?.Title?.Substring(0, Math.Min(Model?.Title?.Length ?? 64, 64));
 }
 
 @section BreadCrumbs {
     <a href="/" class="govuk-back-link">Home</a>
-    <a href="@Url.Action("Details", "Schemes", new { schemeUrl = Model.Scheme.Url })" class="govuk-back-link">@Model.Scheme.Name</a>
+    <a href="@Url.Action("Details", "Schemes", new { schemeUrl = Model?.Scheme?.Url })" class="govuk-back-link">@Model?.Scheme?.Name</a>
+}
+
+@section PreviewErrors {
+    @if ((Model?.Errors != null && Model.Errors.Any()) || string.IsNullOrWhiteSpace(Model?.Scheme?.Url) || string.IsNullOrWhiteSpace(Model?.Title)) {
+        <h3>Errors</h3>
+        <ul>
+            @if (string.IsNullOrWhiteSpace(Model?.Title)) {
+                <li>Title must not be blank</li>
+            }
+            @if (Model?.Scheme == null) {
+                <li>Scheme must be selected before publishing</li>
+            } else if (Model.Scheme.Url == null) {
+                <li>Scheme url cannot be found</li>
+            }
+            @if (Model?.Errors != null) {
```

Comment:
The errors surfaced from the Contentful response provide no value to the users. For example, from the test case study page:

```
Errors
Id: 4I3FDDLhg7tAVuOP5CLvGZ Type: Link LinkType: Entry
Id: 6gfMgPA6rNHg9z2Moq6nxm Type: Link LinkType: Entry
Id: 42ugZj8YTk3YTThGh4KrhZ Type: Link LinkType: Entry
Id: 5d15DvwavKxJAVCdNwXy3D Type: Link LinkType: Entry
Id: 4y7ulNISz3LSOL7CRb9zGO Type: Link LinkType: Entry
Id: 4pILEC7IgjhmbRGf9cMYXb Type: Link LinkType: Entry
Id: qJ5WXof1ShSTOqhPdwSou Type: Link LinkType: Entry
Id: bkfb6QmwqtKxSiAs2cP3B Type: Link LinkType: Entry
```

Therefore, we shouldn't show them to the user (unless there is a way to translate them into meaningful error messages).

That means that we don't have to split up the fetching of preview content in the ContentService, which was needed to only show Contentful error messages related to the content being previewed. (The fetching of a sub-set of the data isn't fine grained enough anyway, it would have to just fetch the content items required for the preview. For example, when reporting errors during previewing a case study page, it fetches all case study pages and all schemes, so the errors displayed could belong to any case study page or scheme, not necessarily the case study being previewed.)

We also have to keep in mind any potential thread safety issues when updating content, although they have been minimised now that updating published content as part of preview has been fixed. The advantage of the current update code is that it has already been proven.

After removing the Contentful errors, that leaves checking for missing mandatory fields. We can achieve that in a much simpler way by checking the model of the preview item, and if there are any issues, display a generic error view (with a generic error model). We then won't require the derived preview pages or PreviewContent (and the unit tests that would have to be written for them) and the existing views wouldn't have to cater for missing mandatory fields.

Reply:
New commits address this